nat/mapping/port_mapping.go

```diff
@@ -86,11 +89,11 @@ func addMapping(m portmap.Interface, protocol string, extPort, intPort int, name
 		log.Debugf("%s Couldn't add port mapping for port %d: %v, retrying with permanent lease", logPrefix, extPort, err)
 		if err := m.AddMapping(protocol, extPort, intPort, name, 0); err != nil {
 			// some gateways support only permanent leases
-			publisher.Publish(traversal.EventTopic, traversal.BuildFailureEvent(err))
+			publisher.Publish(traversal.EventTopic, traversal.BuildFailureEvent(StageName, err))
 			log.Debugf("%s Couldn't add port mapping for port %d: %v", logPrefix, extPort, err)
 			return
 		}
 	}
-	publisher.Publish(traversal.EventTopic, traversal.BuildSuccessEvent())
+	publisher.Publish(traversal.EventTopic, traversal.BuildSuccessEvent(StageName))
```

Comment:
I believe we need not only port mapping events, but hole punching events as well, since we are going to implement some '/connection/id/status' endpoint which should report its nat traversal status to application. This status should source data from such events. 
Trick here is that we are starting to deal with session IDs, maybe session manager should update its status with nat-traversal status reported by events.